dask/array/core.py

```diff
@@ -3187,6 +3224,16 @@ def cov(m, y=None, rowvar=1, bias=0, ddof=None):
         return (dot(X, X.T.conj()) / fact).squeeze()
 
 
+@wraps(np.corrcoef)
+def corrcoef(x, y=None, rowvar=1):
+    c = cov(x, y, rowvar)
+    if c.shape == ():
+        return c/c
+    d = diag(c)
+    d = d.reshape((d.shape[0], 1))
+    return c / sqrt(d * d.T)
```

Comment:
I *think* it would handle it ok, but I'm not confident, especially not in all cases that might arise.  If it does poorly then this problem will run really poorly.

Actually, in this case we might not handle this well just because data transfer is no longer cheap.  Even if we manage to compute related blocks nearby in time it's hard to guarantee that any two independent tasks will compute on the same machine.  The computation of a block of `d * d.T` might very easily occur on a different worker from the corresponding block of `c`.  The relative cost of moving `d` to moving `d * d.T` is large.  

Generally distributed computing has made me wary of creating large intermediate results.